src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -77,10 +91,45 @@ public List<RawEntityTypeSummary> getEntityTypeSummary(Set<UUID> entityTypeIds)
       );
   }
 
+  private List<EntityTypeColumn> fetchColumnNamesForCustomFields(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getEntityTypeDefinition(entityTypeId)
+      .map(EntityType::getSourceView)
+      .orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+    return jooqContext
+      .select(field(REQUIRED_FIELD_NAME), field(REF_ID))
+      .from(sourceViewName)
+      .where(field(WHERE_CLAUSE).eq(CUSTOM_FIELD_TYPE))
+      .fetch()
+      .stream()
+      .map(row -> {
+        Object value = row.get(0);
+        Object extractedRefId = row.get(1);
```

Comment:
Nit: I'd grab these columns by name instead of number. It's more clear and less error-prone

```suggestion
        Object value = row.get(REQUIRED_FIELD_NAME);
        Object extractedRefId = row.get(REF_ID);
```